scheduler-jobs/scheduler-jobs.go

```diff
@@ -0,0 +1,316 @@
+func startServer(ctx context.Context) error {
+	lis, err := net.Listen("tcp", appPort)
+	if err != nil {
+		return fmt.Errorf("failed to listen on %v: %w", appPort, err)
+	}
+
+	s := grpc.NewServer()
+	rtv1.RegisterAppCallbackAlphaServer(s, &appServer{})
+
+	errCh := make(chan error, 1)
+
+	go func() {
+		log.Printf("Starting gRPC server on port %s...\n", appPort)
+		if err := s.Serve(lis); err != nil {
+			errCh <- fmt.Errorf("failed to serve: %w", err)
+		}
+	}()
+
+	select {
+	case <-ctx.Done():
+		log.Println("Context canceled, shutting down gRPC server...")
+		s.GracefulStop()
+		return nil
+	case err := <-errCh:
+		return err
+	}
+}
+
+func (s *appServer) OnBulkTopicEventAlpha1(ctx context.Context, in *rtv1.TopicEventBulkRequest) (*rtv1.TopicEventBulkResponse, error) {
+	return nil, nil
+}
+
+func (s *appServer) OnJobEventAlpha1(ctx context.Context, in *rtv1.JobEventRequest) (*rtv1.JobEventResponse, error) {
+	if strings.HasPrefix(in.GetMethod(), "job/") {
+		if strings.Contains(in.GetMethod(), "oneshot") {
+			count := oneshot.Add(1)
+			if count == maxOneshotTriggerCount {
+				// Reset the oneshot counter
+				oneshot.Store(0)
+				log.Println("Reached max oneshot job count, scheduling new jobs...")
+				// Send signal to start scheduling another batch of one shot jobs
+				oneshotDoneCh <- struct{}{}
+			}
+		} else if strings.Contains(in.GetMethod(), "indefinite") {
+			count := indefinite.Add(1)
+			if count == maxIndefiniteTriggerCount {
+				// Reset the oneshot counter
+				indefinite.Store(0)
+				log.Println("Reached max indefinite job count, scheduling new jobs...")
+				// Send signal to start scheduling another batch of indefinite jobs
+				indefiniteDoneCh <- struct{}{}
+			}
+		} else if strings.Contains(in.GetMethod(), "repeat-job") {
+			count := repeat.Add(1)
+			if count == maxRepeatTriggerCount {
+				// Reset the repeat counter
+				repeat.Store(0)
+				log.Println("Reached max repeat job count, scheduling new jobs...")
+				// Send signal to start scheduling another repeat job
+				repeatDoneCh <- struct{}{}
+			}
+		} else if strings.Contains(in.GetMethod(), "create-delete-job") {
+			count := createDelete.Add(1)
+			log.Printf("create-delete-job triggered count: %d\n", count)
+
+			if count == maxCreateDeleteTriggerCount {
+				log.Println("Received the single create-delete-job, deleting it...")
+				// Send signal to delete the job
+				receivedSingleJobDoneCh <- struct{}{}
+			} else {
+				log.Printf("Received too many single repeat, create-delete-job jobs. Count: %d...\n", createDelete.Load())
+			}
+		}
+	}
+	return nil, nil
+}
+
+func scheduleOneshotJobs(daprClient client.Client) {
+	for i := 0; i < 100; i++ {
+		name := "oneshot-job-" + strconv.Itoa(i)
+		req := &client.Job{
+			Name:     name,
+			Schedule: "@every 30s",
+			Repeats:  1, // one shot job
+			DueTime:  "5s",
+			TTL:      "40s",
+			Data:     nil,
+		}
+		err := daprClient.ScheduleJobAlpha1(context.Background(), req)
```

Comment:
Can we add timeouts in the ctx of the schedule/delete functions pls. We should also be checking if the main context is still alive in the loop.